### PR TITLE
doc/releases/release-notes-3.5: add RISC-V release notes

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR adds RISC-V release notes for Zephyr v3.5.0.

### CODE REVIEW

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -45,6 +45,18 @@ Architectures
 
 * RISC-V
 
+  * Added support for detecting null pointer exception using PMP.
+  * Renamed :kconfig:option:`CONFIG_RISCV_MTVEC_VECTORED_MODE` to
+    :kconfig:option:`CONFIG_RISCV_VECTORED_MODE`.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:50]: this might be better suited for the migration guide? /cc @carlescufi

Comment (fkokosinski) [reply doc/releases/release-notes-3.5.rst:50]: Hi! Judging from the migration guide's content, it indeed looks like the highlighted point should be added there. Should I remove it from release notes as well?

Comment (jhedberg) [reply doc/releases/release-notes-3.5.rst:71]: > Hi! Judging from the migration guide's content, it indeed looks like the highlighted point should be added there. Should I remove it from release notes as well?

Yes, as per guidance from @carlescufi we shouldn't duplicate information in the two files

Comment (fkokosinski) [reply doc/releases/release-notes-3.5.rst]: Ok. I've removed it from the release notes and now this point is only present in the migration guide.

**Review by jhedberg — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@fkokosinski could you rebase please?

**Comment by fkokosinski** [comment]
> @fkokosinski could you rebase please?

Done

**Review by kartben — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**